esp-hal/src/i2c/master/mod.rs

```diff
@@ -2409,8 +2360,8 @@ impl Driver<'_> {
         }
 
         self.start_read_operation(address, buffer, start, will_continue)?;
-        self.read_all_from_fifo(buffer).await?;
         self.wait_for_completion(true).await?;
+        self.read_all_from_fifo(buffer)?;
```

Comment:
This one line swap is the real fix for the "blocking in async driver" issue, right?

Reply:
yes - this would have been enough already